File: Hallowed_by_thy_name_java/src/main/java/com/xerox/project/HALLOWED/BY/THY/NAME/services/ProductService.java
package com.xerox.project.HALLOWED.BY.THY.NAME.services;

import com.xerox.project.HALLOWED.BY.THY.NAME.domain.product.ProductModel;
import com.xerox.project.HALLOWED.BY.THY.NAME.repositories.ProductRepository;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class ProductService {
    private final ProductRepository productRepository;

    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    public Mono<ProductModel> post (ProductModel productModel){
        return productRepository.save(productModel);
    }
}

File: lesgooooooo_go/services/product_service.go
package services

import "raiox/main/services"

type ProductService struct {
	productRepository *services.ProductRepository
}

func NewProductService(productRepository *services.ProductRepository) *ProductService {
	return &ProductService{productRepository: productRepository}
}

func (s *ProductService) Post(productModel *Pr) (*product.ProductModel, error) {
	createdProduct, err := s.productRepository.Save(productModel)
	if err != nil {
		return nil, err
	}

	return createdProduct, nil
}
